golang/cosmos/x/swingset/abci.go

```diff
@@ -14,21 +14,27 @@ import (
 )
 
 type beginBlockAction struct {
-	Type        string       `json:"type"`
+	Type        string       `json:"type" default:"BEGIN_BLOCK"`
 	BlockHeight int64        `json:"blockHeight"`
 	BlockTime   int64        `json:"blockTime"`
```

Comment:
Here and below, consider embedding a common `ActionHeader` structure as in Richard's version. Also, make some reference to the understanding that certain fields will be automagically populated. Having the common header lets you put that reference in just one place.